src/main/java/io/swagger/api/watchlist/WatchlistApiController.java

```diff
@@ -82,7 +88,18 @@ public ResponseEntity<Watchlist> watchlistWatchlistIdVisibilityPut(@Parameter(in
             return ResponseEntity.notFound().build();
         }
 
+        AddHateoasUrls(watchlist);
         return ResponseEntity.ok(watchlist);
     }
 
+    private void AddHateoasUrls(Watchlist watchlist)
+    {
+        watchlist.add(linkTo(methodOn(WatchlistApiController.class).watchlistWatchlistIdGet(watchlist.getId())).withSelfRel());
+        watchlist.add(linkTo(methodOn(WatchlistApiController.class).watchlistWatchlistIdVisibilityPut(watchlist.getId(), new WatchlistVisiblity())).withRel("set-watchlist-visibility"));
+
+        for (MediaItem mediaItem : watchlist.getMediaItems()) {
+            mediaItem.add(linkTo(methodOn(MediaApiController.class).mediaMediaIdGet(mediaItem.getId())).withSelfRel());
```

Comment:
I'm open to calling this the same thing as your API: `get-media-metadata`. "Self" does make sense in that this is a call to get the Media Item. However, MediaItem only has the Id and Name. So, "Self" may not fully make sense. Anyway, I'm not sure which to use.

Reply:
I think Self makes sense since it's not technically a call to get media metadata